pwa-devdocs/_drafts/configuration-management/buildpack-cli/load-env/index.md

```diff
@@ -0,0 +1,192 @@
+// If the current file is in project root, use the Node builtin `__dirname`.
+const configuration = loadEnvironment('/Users/me/path/to/project');
+
+// `loadEnvironment` has now read the contents of
+// `/Users/me/path/to/project/.env` and merged it with any environment
+// variables that were alredy set.
+
+// Create an UPWARD server using env vars that begin with `UPWARD_JS_`
+createUpwardServer(configuration.section('upwardJs'));
+
+// If these environment variables are set:
+//
+// UPWARD_JS_HOST=https://local.upward/
+// UPWARD_JS_PORT=8081
+//
+// then `configuration.section('upwardJs')` produces this object:
+//
+// {
+//   host: 'https://local.upward',
+//   port: '8081'
+// }
+//
+// No other environment variables are included in this object unless they begin
+// with `UPWARD_JS_` which is the equivalent of `upwardJs` camel-cased.
+
+
+// The .all() method turns the whole environment into an object, with all
+// CONSTANT_CASE names turned into camelCase names.
+const allConfig = configuration.all();
+
+// This object will have one property for each set environment variable,
+// including the UPWARD variables named above.
+// But `configuration.all()` does not namespace them, they have longer names:
+//
+// {
+//   upwardJsHost: 'https://local.upward',
+//   upwardJsPort: '8081'
+// }
+//
+// This huge object defeats the purpose of loadEnvironment() and should
+// only be used for debugging.
+
+// Instead, let's create an UPWARD server combining two environment variable
+// sections with hardcoded overrides to some values.
+createUpwardServer({
+  ...configuration.section('upwardJs'),
+  ...configuration.section('magento'),
+  bindLocal: true
+});
+
+// This uses JavaScript object spreading to combine several sections of
+// configuration and override a value.
+// If the environment contains these values:
+//
+// UPWARD_JS_HOST=https://local.pwadev
+// UPWARD_JS_PORT=443
+// UPWARD_JS_BIND_LOCAL=
+// MAGENTO_BACKEND_URL=https://local.magento
+//
+// Then the above code passes the following object to `createUpwardServer`:
+//
+// {
+//   host: 'https://local.pwadev',
+//   port: '443',
+//   backendUrl: 'https://local.magento',
+//   bindLocal: true
+// }
+
+
+// The `sections()` method can split an env object into named subsections:
+createUpwardServer(configuration.sections('upwardJs', 'magento'));
+
+// Given the same environment variables as above, this code will pass the
+// following to `createUpwardServer`:
+//
+// {
+//   upwardJs: {
+//     host: 'https://local.pwadev',
+//     port: '443',
+//     bindLocal: '' // the null string is used as a falsy value
+//   },
+//   magento: {
+//     backendUrl: 'https://local.magento'
+//   }
+// }
+//
+// (The above is not the actual config object format for `createUpwardServer`,
+// but if it was, that's how you'd make it.)
+
+// Use the convenience properties `isProd` and `isDev` instead of testing
+// `process.env.NODE_ENV` directly:
+if (configuration.isDev) {
```

Comment:
Above we reference `isDevelopment` and `isProduction` but here you mention the shorthand. Which is correct?

Reply:
They are both correct. If you look at the reference table, the shorthand versions are under the `Alias` column.